File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        window = set()
        left = 0
        res = 0

        for i, char in enumerate(s):
            while char in window:
                window.remove(s[left])
                left += 1
            window.add(char)
            res = max(res, i- left + 1)
        
        return res


        

File: Data Structures & Algorithms/concatenation-of-array/submission-2.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        //init 
        int n = nums.length;
        int[] res = new int[n*2];
        
        // main
        for(int i = 0; i < n; i++){
            res[i] = nums[i];
            res[i+n] = nums[i];
        }

        return res;
    }
}